x-pack/platform/test/functional/page_objects/onechat_page.ts

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { last } from 'lodash';
+
+import type { FtrProviderContext } from '../ftr_provider_context';
+import { FtrService } from '../ftr_provider_context';
+import type { LlmProxy } from '../../onechat_api_integration/utils/llm_proxy';
+import { toolCallMock } from '../../onechat_api_integration/utils/llm_proxy/mocks';
+
+export class OneChatPageObject extends FtrService {
+  private readonly testSubjects = this.ctx.getService('testSubjects');
+  private readonly retry = this.ctx.getService('retry');
+  private readonly browser = this.ctx.getService('browser');
+  private readonly common = this.ctx.getPageObject('common');
+
+  constructor(ctx: FtrProviderContext) {
+    super(ctx);
+  }
+
+  /**
+   * Navigate to the OneChat app
+   */
+  async navigateToApp(path: string = 'conversations/new') {
+    await this.common.navigateToApp('agent_builder', { path });
+  }
+
+  /**
+   * Type a message in the conversation input
+   */
+  async typeMessage(message: string) {
+    const inputElement = await this.testSubjects.find('onechatAppConversationInputFormTextArea');
+    await inputElement.click();
+    await inputElement.type(message);
+  }
+
+  /**
+   * Send the current message
+   */
+  async sendMessage() {
+    const sendButton = await this.testSubjects.find('onechatAppConversationInputFormSubmitButton');
+    await sendButton.click();
+  }
+
+  /**
+   * Extract conversation ID from the current browser URL
+   */
+  async getCurrentConversationIdFromUrl(): Promise<string> {
+    return await this.retry.try(async () => {
+      const url = await this.browser.getCurrentUrl();
+      // URL should be something like: /app/agent_builder/conversations/{conversationId}
+      const match = url.match(/\/conversations\/([^\/\?]+)/);
+      if (!match) {
+        throw new Error('Could not extract conversation ID from URL');
+      }
+      return match[1];
+    });
+  }
```

Comment:
I wish there was an easier way to do this! But we don't expose the `conversationId` in the UI besides the browser URL AFAIK 🤔